### PR TITLE
add regresssion_samples explanation

### PR DESCRIPTION
**Pull request by sashashura**

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -5,6 +5,7 @@ repos:
     rev: v4.5.0
     hooks:
       - id: trailing-whitespace
+        args: [--markdown-linebreak-ext=md]
```
Comment (seladb) [inline .pre-commit-config.yaml:8]: Why is this change needed?

Comment (sashashura) [reply .pre-commit-config.yaml:8]: Because two spaces at the end is a valid markdown line-break syntax. Pre-commit was too aggressive for markdown files.

Comment (seladb) [reply .pre-commit-config.yaml:8]: While two spaces at the end is a valid markdown line-break, it's not considered a best practice: https://www.markdownguide.org/basic-syntax/#line-break-best-practices

Comment (sashashura) [reply .pre-commit-config.yaml]: Hmm, The best practice is to mix html with markdown?? Just saying - https://grep.app/search?q=--markdown-linebreak-ext
Anyway, as you wish, I have reverted it.

**Review by seladb — APPROVED** [review]
